### ISSUE TITLE
Check a tuple for containing a mutable object

### ISSUE THREAD

**Issue by KonsKo** [wemake-services/wemake-python-styleguide#3090]
# Rule request

Hello,
I would like to suggest a new rule. I could not find it in the existing rules, but if it exists so, sorry for duplication.

## Thesis

I am proposing a new check for a tuple if a tuple contains a mutable object.  

```
Python 3.12.7 (...) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
mutable_variable: list = [1, 2, 3]
tuple_variable: tuple = (1, 2, 3, mutable_variable)
tuple_variable
(1, 2, 3, [1, 2, 3])
mutable_variable[0] = 'new_val'
tuple_variable
(1, 2, 3, ['new_val', 2, 3])
```

## Reasoning

It is dangerous to use tuples with mutable objects. And I think it may help many people to understand it.
We expects immutable behavior from a tuple.
But we know that for container-like sequences (for our case for a tuple), all elements are held by reference.
When we put a mutable container-like sequence into a tuple, and after that we change value of it, tuple will be changed.
It may look like unexpected behavior and may lead to errors.

**Comment by sobolevn** [comment]
Good idea, thank you!

**Comment by sobolevn** [comment]
Turns out that a lot of API rely on that: django settings, pytest fixtures, etc.

Sadly, I cannot make it good enough not to raise a lot of false-positives.
